agent/grpc-external/services/peerstream/subscription_manager.go

```diff
@@ -664,6 +664,18 @@ func createDiscoChainHealth(
 	}
 }
 
+var statusScores = map[string]int{
+	// 0 is reserved for unknown
+	api.HealthMaint:    1,
+	api.HealthCritical: 2,
+	api.HealthWarning:  3,
+	api.HealthPassing:  4,
+}
+
+func isStatusBetter(curr, next string) bool {
```

Comment:
Nit. To simplify mental-effort, I would have this function return the winning string instead of having it return a bool. Also, the name "better" makes me think that it's returning the healthier status, which is a little confusing. I'd probably name it `getMostImportantStatus()` or something similar.